web/modules/custom/joinup_event/joinup_event.module

```diff
@@ -5,49 +5,29 @@
  * Contains joinup_event.module.
  */
 
-use \Drupal\core\Form\FormStateInterface;
+use Drupal\Core\Entity\EntityTypeInterface;
 
 /**
- * Implements hook_form_FORM_ID_alter().
+ * Implements hook_entity_bundle_field_info_alter().
  */
-function joinup_event_form_node_event_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  $form['#validate'][] = 'joinup_event_form_default_validate';
-}
-
-/**
- * Implements hook_form_FORM_ID_alter().
- */
-function joinup_event_form_node_event_edit_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  $form['#validate'][] = 'joinup_event_form_default_validate';
+function joinup_event_entity_bundle_field_info_alter(&$fields, EntityTypeInterface $entity_type, $bundle) {
+  if ($entity_type->id() == 'node' && $bundle == 'event') {
+    if (!empty($fields['field_event_short_title'])) {
+      $fields['field_event_short_title']->addConstraint('UniqueField');
+    }
+    if (!empty($fields['title'])) {
+      $fields['title']->addConstraint('UniqueField');
+    }
+  }
 }
 
 /**
- * Validates the default joinup event form.
- *
- * The validation includes the the short title.
- * The short title field is mandatory and it has to be unique
- * in a global scope among events.
- *
- * @param array $form
- *   The form array.
- * @param \Drupal\core\Form\FormStateInterface $form_state
- *   The form state object.
+ * Implements hook_entity_base_field_info_alter().
  */
-function joinup_event_form_default_validate(&$form, FormStateInterface $form_state) {
-  // Check the short title.
-  // Load all document entities with this short title.
-  /** @var \Drupal\Core\Entity\EntityInterface $entity */
-  $entity = $form_state->getFormObject()->getEntity();
-
-  $query = \Drupal::entityQuery('node')
-    ->condition('type', 'event')
-    ->condition('field_event_short_title', $form_state->getValue('field_event_short_title')[0]['value']);
-  if ($entity->id()) {
-    $query->condition('nid', $entity->id(), '<>');
-  }
-  $results = $query->execute();
-
-  if (!empty(reset($results))) {
-    $form_state->setErrorByName('field_event_short_title', t('There is already an event with this short title.'));
+function joinup_event_entity_base_field_info_alter(&$fields, EntityTypeInterface $entity_type) {
+  if ($entity_type->id() == 'node') {
+    if (!empty($fields['field_event_short_title'])) {
+      $fields['field_event_short_title']->addConstraint('UniqueField');
+    }
```

Comment:
Is this still needed now that the constraint is already added to the short title in `joinup_event_entity_bundle_field_info_alter()`?